src/nlp.jl

```diff
@@ -236,9 +236,40 @@ end
 Set the value of the MOI attribute [`MOI.NLPBlockDualStart`](@ref).
 
 The start vector corresponds to the Lagrangian duals of the nonlinear
-constraints, in the order given by [`all_nl_constraints`](@ref).
+constraints, in the order given by [`all_nl_constraints`](@ref). That is, you
+must pass a single start vector corresponding to all of the nonlinear
+constraints in a single function call; you cannot set the dual start value of
+nonlinear constraints one-by-one. The example below demonstrates how to use
+[`all_nl_constraints`](@ref) to create a mapping between the nonlinear
+constraint references and the start vector.
 
 Pass `nothing` to unset a previous start.
+
+## Examples
+
+```jldoctest
+julia> model = Model();
+
+julia> @variable(model, x[1:2]);
+
+julia> nl1 = @NLconstraint(model, x[1] <= sqrt(x[2]));
+
+julia> nl2 = @NLconstraint(model, x[1] >= exp(x[2]));
+
+julia> start = Dict(nl1 => -1.0, nl2 => 1.0);
+
+julia> start_vector = [start[con] for con in all_nl_constraints(model)]
```

Comment:
Do you think it would be simpler to say: 
```
start_vector = zeros(num_nl_constraints(model))
start_vector[nl1.value] = -1
start_vector[nl2.value] = 1
```

Reply:
I generally shied away from theses `.value` accesses. I also don't know if they're part of the public API, which documenting like this would suggest.

In general, the user will need to maintain a separate mapping from constraint -> start value, so they'd probably write something like
```julia
for (con, start) in dict
    start_vector[con.value] = start
end
```
anyway. I don't see anyone hard-coding the start values like you have.

Reply:
Ok, sounds good.